src/socket/delegates.cr

```diff
@@ -0,0 +1,180 @@
+module Socket
+  # :nodoc:
+  macro delegate_close
```

Comment:
some of these macros could be modules that are included. If so, I'm ok with reusing code using that artifact.

Reply:
Yeah, maybe. But I'm not sure it makes much sense. It doesn't hurt to just duplicate those methods (it's typically only two or three uses for each). I would only use a module if it communicates some sort of common behaviour that could be useful as an individual trait.

Maybe the IO delegator methods could be put in a `IO::Delegator` module. This might actually be reusable in other places (`HTTP::Server::Response`, `HTTP::UnknownLengthContent`, or `IO::Hexdump` come to mind).